src/PackageCompiler.jl

```diff
@@ -786,4 +941,16 @@ function bundle_artifacts(ctx, app_dir)
     return
 end
 
+function bundle_headers(dest_dir, header_files)
+    isempty(header_files) && return
+    include_dir = joinpath(dest_dir, "include")
+    mkpath(include_dir)
+
+    for header_file in header_files
+        new_file = joinpath(include_dir, basename(header_file))
```

Comment:
This assumes all header files are at the top level.  For basic use cases, this is probably fine, but it might be good to allow paths relative to the top level include.  Not sure of the best way to do this...